src/Surface.zig

```diff
@@ -3870,22 +3870,35 @@ pub fn performBindingAction(self: *Surface, action: input.Binding.Action) !bool
         },
 
         .copy_to_clipboard => {
-            // We can read from the renderer state without holding
-            // the lock because only we will write to this field.
-            if (self.io.terminal.screen.selection) |sel| {
-                const buf = self.io.terminal.screen.selectionString(self.alloc, .{
-                    .sel = sel,
-                    .trim = self.config.clipboard_trim_trailing_spaces,
-                }) catch |err| {
-                    log.err("error reading selection string err={}", .{err});
-                    return true;
-                };
+            if (copyToClipboard(self)) {
+                return true;
+            }
+        },
+
+        .copy_or_interrupt => {
+            if (copyToClipboard(self)) {
+                return true;
+            } else {
+                const buf = try self.alloc.alloc(u8, 1);
                 defer self.alloc.free(buf);
 
-                self.rt_surface.setClipboardString(buf, .standard, false) catch |err| {
-                    log.err("error setting clipboard string err={}", .{err});
-                    return true;
-                };
+                buf[0] = 3;
+
+                self.io.queueMessage(try termio.Message.writeReq(
+                    self.alloc,
+                    buf,
+                ), .unlocked);
+
```

Comment:
It would be slightly more efficient to directly use a small write request since we're only sending a single byte. See an example here:

https://github.com/ghostty-org/ghostty/blob/3059d9036b233d672e6dc7cdfb91c822978cc1ef/src/Surface.zig#L2646-L2658

Reply:
Im not sure which I need .locked vs .unlocked ive it left as unlocked